fixed fire playing hurt animation and made regen checks more sensitive

Jackiecrazy opened the pull request:
LivingHurtEvent is called before armor and damage calculations, so if you have mods that modify armor values (such as my own Armor Curve), or modify health values to be very low (such as Spice of Life: Carrot Edition), or just are taking a huge hit, you will be forced into regen early. Likewise, though, it's too late to cancel hurt animations at that point. I moved hurt canceling to LivingAttackEvent so the hurt is never processed and moved regen checking to the very bottom of LivingDeath, (presumably) after every other mod checks their anti-death measures already, including totems of undying. Those are the only changes.

Jackiecrazy committed:
fixed event problems

Jackiecrazy committed:
Update RegenEventHandler.java

Swirtzly commented:
Looks good, have you tested it?

Jackiecrazy commented:
I have, and it works as expected.

Darkmega18 commented:
I haven't been slapped with an out of place one shot I don't think, but this'll probably be a fix to my issue with the regenerations and death preventionary stuff not being able to do it's job before regens. Although now it just needs ability to save you from critical death and have your living status be reset properly and not cause regens to be deleted if you didn't actually die.

Jackiecrazy committed:
fixed tanking death keeping select effects from critical phase and resetting DNA, fixed derped critical damage application and made kill on regen work in creative, added chance to Lindos spawn

Jackiecrazy commented:
I put some more changes to prevent health, speed, and shader weirdness if someone were to forcibly prevent death with, say, resistance V during critical phase. I also changed the coin flip to spawn a Lindos particle in post regen phase to percentage chance. Since I don't know French, Russian or Swedish, I only updated the English file for that option. As well, there's a slight problem with critical kills, being that you set status before dealing damage, which means that it'll always deal the non-piercing critical_kill damage source, which destroys all armor you're wearing. I'm assuming this is unintended, so I moved status update after damage dealing and set critical_kill to work in creative as well.

Jackiecrazy committed:
added lindos burping, decreased bio-container efficiency

Jackiecrazy committed:
did that say 5? gotta crank up those numbers, those are rookie numbers.

Jackiecrazy commented:
newest commit adds a configurable chance to burp lindos particles during post-regen phase like the 10th doctor, and decreases bio-container marginal utility and range. These are more bells&whistles than bug fixes, feel free to reject them if you wish.

Jackiecrazy committed:
forgot to update lang!

Darkmega18 commented:
diggity. you should hire this guy Swirtz. :O

Swirtzly commented:
@Jackiecrazy I like what you have done so far, is there anything else you wish to add or is this the right time to be merged?

Jackiecrazy committed:
fixed bio-containers only checking around spawn

Jackiecrazy commented:
grah, that's what you get for coding at 3AM without debugging, fixed the last issue >_>
I'm pretty much just going through the code and changing/fixing what I can; I wouldn't really know what to add even if you asked me, since I'm generally very careful with other people's projects. It's a good time to merge, I'd say.

Swirtzly commented:
Released here: https://www.curseforge.com/minecraft/mc-mods/regeneration/files/2971156


If you want to port these changes to 1.14, I would appreciate it, but you don't need to
